cert/sign.go

```diff
@@ -76,15 +76,89 @@ func (t *TBSCertificate) Sign(signer Certificate, curve Curve, key []byte) (Cert
 	}
 }
 
+// readyToSign checks all signing requirements that don't require us to cross-reference with a CA
+func (t *TBSCertificate) readyToSign() error {
+	if len(t.PublicKey) == 0 {
+		return fmt.Errorf("public key not set")
+	}
+
+	if !t.IsCA && len(t.Networks) == 0 {
+		return fmt.Errorf("non-CA certificates must contain at least one network")
+	}
+
+	hasV4Networks := false
+	hasV6Networks := false
+	for _, n := range t.Networks {
+		if !n.IsValid() || !n.Addr().IsValid() {
+			return fmt.Errorf("invalid network: %s", n)
+		}
+		if t.Version == Version1 && n.Addr().Is6() {
+			return fmt.Errorf("certificate v1 may not contain IPv6 networks: %v", t.Networks)
+		}
+		if n.Addr().Zone() != "" {
+			return fmt.Errorf("networks may not contain zones: %s", n)
+		}
+		if n.Addr().Is4In6() {
+			return fmt.Errorf("4in6 networks are not allowed: %s", n)
+		}
+		if !t.IsCA && n.Addr().IsUnspecified() {
+			return fmt.Errorf("non-CA certificates must not use the zero address as a network: %s", n)
+		}
+
+		hasV4Networks = hasV4Networks || n.Addr().Is4()
+		hasV6Networks = hasV6Networks || n.Addr().Is6()
+	}
+
+	slices.SortFunc(t.Networks, comparePrefix)
+	err := findDuplicatePrefix(t.Networks)
+	if err != nil {
+		return err
+	}
+
+	for _, n := range t.UnsafeNetworks {
+		if !n.IsValid() || !n.Addr().IsValid() {
+			return fmt.Errorf("invalid unsafe network: %s", n)
+		}
+		if n.Addr().Zone() != "" {
+			return fmt.Errorf("unsafe_networks may not contain zones: %s", n)
+		}
+		//todo are unsafe networks that overlap networks allowed?
```

Comment:
Yes, you might have `10.0.0.0/8` point to one host but `10.0.0.0/16` to another, it makes less sense to have overlapping in a CA certificate but that's also fine.